dialogue-core/src/main/java/com/palantir/dialogue/core/StickyConcurrencyLimitedChannel.java

```diff
@@ -49,10 +51,9 @@ final class StickyConcurrencyLimitedChannel implements LimitedChannel {
     @Override
     public Optional<ListenableFuture<Response>> maybeExecute(
             Endpoint endpoint, Request request, LimitEnforcement limitEnforcement) {
-        Optional<CautiousIncreaseAggressiveDecreaseConcurrencyLimiter.Permit> maybePermit =
-                limiter.acquire(limitEnforcement);
-        if (maybePermit.isPresent()) {
-            CautiousIncreaseAggressiveDecreaseConcurrencyLimiter.Permit permit = maybePermit.get();
+        @Nullable Permit maybePermit = limiter.acquire(limitEnforcement);
+        if (maybePermit != null) {
+            CautiousIncreaseAggressiveDecreaseConcurrencyLimiter.Permit permit = maybePermit;
```

Comment:
do we need this local var? I'd suggest removing it completely or if you want to keep it move it inside the else block scope

```suggestion
```